cm-ovirt-vm-creator.py

```diff
@@ -0,0 +1,193 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+# cm-ovirt-vm-creator.py - Creates a set of VMs to be used by cm-jenkins as openshift nodes 
+#
+# Copyright © 2018 Red Hat Inc.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+from __future__ import print_function, unicode_literals
+import os
+import sys
+import traceback
+import argparse
+import logging
+import time
+import ovirtsdk4 as sdk
+import ovirtsdk4.types as types
+
+
+
+
+def str2bool(s):
+    """ Convert str argument to bool """
+    if s.lower() in ('', 'yes', 'true', 't', 'y', '1'):
+        return True
+    elif s.lower() in ('no', 'false', 'f', 'n', '0'):
+        return False
+    else:
+        raise argparse.ArgumentTypeError('Boolean value expected.')
+
+
+
+logging.basicConfig(level=logging.DEBUG, filename="ocp-on-rhev-deployer.log")
+
+
+
+def doWork(args):
+    connection = None
+    try:
+        connection = sdk.Connection(url=args.ovirt_url , 
+                    username=args.ovirt_user, 
+                    password=os.environ[args.ovirt_pass], 
+                    ca_file=args.ovirt_ca_pem_file , 
+                    debug=True, 
+                    log=logging.getLogger())
+        system_service = connection.system_service()
+        vms_service = system_service.vms_service()
+        cluster_nodes = []
+        vm_name_template="{name_prefix}-{node_type}{i:03d}"
+        for idx in range(1, args.masters+1):
+            cluster_nodes.append(vm_name_template.format(name_prefix=args.name_prefix, node_type="master", i=idx))
+        for idx in range(1, args.infra_nodes+1):
+            cluster_nodes.append(vm_name_template.format(name_prefix=args.name_prefix, node_type="infra", i=idx))
+        for idx in range(1, args.nodes+1):
+            cluster_nodes.append(vm_name_template.format(name_prefix=args.name_prefix, node_type="compute", i=idx))
+        print (cluster_nodes)
+        if args.info:
+            getVmsInfo(connection, system_service, vms_service, cluster_nodes, args)
+        else:
+            createVMs(connection, system_service, vms_service, cluster_nodes, args)
+    except:
+        traceback.print_exc()
+    finally:
+        connection.close()
```

Comment:
Probably best to use `if connection is not None:` here, in case the failure was in creating the connection object (for example, if the pem file path was wrong)